src/mapdata.cpp

```diff
@@ -477,7 +477,11 @@ ter_id t_null,
     // Temple tiles
     t_rock_red, t_rock_green, t_rock_blue, t_floor_red, t_floor_green, t_floor_blue,
     t_switch_rg, t_switch_gb, t_switch_rb, t_switch_even, t_open_air, t_plut_generator,
-    t_pavement_bg_dp, t_pavement_y_bg_dp, t_sidewalk_bg_dp, t_guardrail_bg_dp;
+    t_pavement_bg_dp, t_pavement_y_bg_dp, t_sidewalk_bg_dp, t_guardrail_bg_dp,
+    // Railroad
+	t_rubble,
+    t_railroad_track, t_railroad_track_on_tie, t_buffer_stop_h, t_buffer_stop_v, t_railroad_tie, t_railroad_tie_h, t_railroad_tie_v, t_railroad_tie_d1, t_railroad_tie_d2, t_railroad_crossing_signal, t_crossbuck_wood, t_crossbuck_metal
```

Comment:
Needing to define those is a big red flag that it isn't all done as clearly as it could and should be.

Reply:
Some of this is currently being used in the CPP mapgen code. I will try to move most of the mapgen to json and leave only those things in the CPP which are really necessary.